drgn/helpers/common/memory.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Memory
+------
+
+The ``drgn.helpers.memory`` module provides helpers for working with memory and addresses.
+"""
+
+import operator
+from typing import Optional, Union, overload
+
+import drgn
+from drgn import IntegerLike, Object, Program, SymbolKind
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.slab import find_containing_slab_cache
+
+__all__ = ("identify_address",)
+
+
+_SYMBOL_KIND_STR = {
+    SymbolKind.OBJECT: "object symbol",
+    SymbolKind.FUNC: "function symbol",
+}
+
+
+@overload
+def identify_address(addr: Object) -> Optional[str]:
+    """"""
+    ...
+
+
+@overload
+def identify_address(prog: Program, addr: IntegerLike) -> Optional[str]:
+    """
+    Try to identify what an address refers to.
+
+    For all programs, this will identify addresses as follows:
+
+    * Object symbols (e.g., addresses in global variables):
+      ``object symbol: {symbol_name}+{hex_offset}``.
+    * Function symbols (i.e., addresses in functions):
+      ``function symbol: {symbol_name}+{hex_offset}``.
+    * Other symbols: ``symbol: {symbol_name}+{hex_offset}``.
+
+    Additional, for the Linux kernel, this will identify:
```

Comment:
Oops, I made a typo here (this should be "Additionally"). I can fix it on my end.